File: backend/app/schemas/artifact.py
"""
ContentForge AI -- Artifact & Verification Schemas
"""

from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel, Field
from app.schemas.enums import ArtifactStatus, VerificationStatus


class ProvenanceInfo(BaseModel):
    """Cryptographic provenance ledger status for a finalized artifact."""
    status: str  # NONE | PENDING | ANCHORED
    reference: Optional[str] = None
    artifact_hash: Optional[str] = None
    verification_hash: Optional[str] = None
    ledger_tx_id: Optional[str] = None
    anchored_at: Optional[str] = None


class ArtifactResponse(BaseModel):
    artifact_id: str
    transformation_request_id: Optional[str] = None
    cco_version_id: Optional[str] = None
    type: str  # presentation, executive_summary, advisory, etc.
    version: int = 1
    parent_artifact_id: Optional[str] = None
    status: ArtifactStatus = ArtifactStatus.GENERATING
    filename: Optional[str] = None
    download_url: Optional[str] = None
    checksum: Optional[str] = None
    template_config: Optional[dict[str, Any]] = None
    render_error: Optional[str] = None
    content_json: dict[str, Any] = {}
    verification: dict[str, Any] = {}
    provenance: Optional[ProvenanceInfo] = None
    created_at: Optional[datetime] = None


class ArtifactVersionResponse(BaseModel):
    artifact_id: str
    version: int
    status: ArtifactStatus
    checksum: Optional[str] = None
    download_url: Optional[str] = None
    created_at: Optional[datetime] = None
    parent_artifact_id: Optional[str] = None


class ArtifactVerificationResponse(BaseModel):
    artifact_id: str
    status: VerificationStatus
    grounding_score: float
    consistency_score: float
    unsupported_claim_count: int
    issues: list[dict[str, Any]] = []


class ArtifactReviseRequest(BaseModel):
    instructions: str = Field(..., min_length=3, description="Revision feedback/prompt adjustments")
    target_sections: Optional[list[str]] = Field(None, description="Specific sections to modify")


class ArtifactFinalizeRequest(BaseModel):
    notes: Optional[str] = Field(None, description="Optional finalization notes or owner signature")

File: frontend/src/types/artifact.ts
import { ArtifactStatus, VerificationStatus } from "./enums";

export interface SlideData {
  slide_number: number;
  title: string;
  key_message: string;
  body: string[];
  speaker_notes?: string;
  evidence_refs?: string[];
}

export interface VerificationIssueItem {
  id?: string;
  claim?: string;
  status?: "supported" | "flagged" | "unsupported" | string;
  severity?: "HIGH" | "MEDIUM" | "LOW" | string;
  category?: string;
  location?: string;
  offending_text?: string;
  suggested_fix?: string;
  evidence_id?: string;
  reason?: string;
  evidence_ref?: string;
}

export interface VerificationReport {
  status: VerificationStatus | "PASSED" | "REVISION_REQUIRED" | "FAILED" | string;
  grounding_score: number;
  consistency_score: number;
  unsupported_claim_count: number;
  citation_coverage?: number;
  issues?: VerificationIssueItem[];
  unsupported_claims?: string[];
}

export interface ProvenanceInfo {
  /** NONE = never finalized, PENDING = finalized awaiting anchor, ANCHORED = ledger confirmed */
  status: "NONE" | "PENDING" | "ANCHORED" | string;
  reference?: string;         // ProvenanceRecord UUID e.g. PRV-XXXXXXXX
  artifact_hash?: string;     // SHA-256 of stored binary deliverable
  verification_hash?: string; // SHA-256 of verification report JSON
  ledger_tx_id?: string;      // Hyperledger Fabric transaction ID (PENDING until confirmed)
  anchored_at?: string;       // ISO-8601 timestamp of anchoring
}

export interface ArtifactVersionItem {
  artifact_id: string;
  version: number;
  status: ArtifactStatus | string;
  checksum?: string;
  download_url?: string;
  created_at?: string;
  parent_artifact_id?: string;
}

export interface ArtifactItem {
  artifact_id: string;
  transformation_request_id: string;
  cco_version_id: string;
  type: string;
  version: number;
  parent_artifact_id?: string;
  status: ArtifactStatus | string;
  filename: string;
  download_url?: string;
  checksum?: string;
  storage_key?: string;
  template_id?: string;
  template_config?: Record<string, any>;
  render_error?: string;
  classification?: string;
  content_json: Record<string, any>;
  verification?: VerificationReport;
  provenance?: ProvenanceInfo;
  created_at?: string;
  available_formats?: string[];
}
